homeassistant/core.py

```diff
@@ -373,6 +374,36 @@ def from_dict(cls, json_dict):
         return cls(json_dict['entity_id'], json_dict['state'],
                    json_dict.get('attributes'), last_changed, last_updated)
 
+    def as_number(self):
+        """Try to coerce our state to a number.
+
+        Raises ValueError if this is not possible.
+        """
+        # Avoid circular import
+        from homeassistant.components.sun import (STATE_ABOVE_HORIZON,
+                                                  STATE_BELOW_HORIZON)
+
+        if self.state in (STATE_ON, STATE_LOCKED, STATE_ABOVE_HORIZON,
+                          STATE_OPEN):
+            return 1
+        elif self.state in (STATE_OFF, STATE_UNLOCKED, STATE_UNKNOWN,
+                            STATE_BELOW_HORIZON, STATE_CLOSED):
+            return 0
+        elif self.state is None:
+            return 0
+        elif isinstance(self.state, bool):
+            return int(self.state)
```

Comment:
Somewhere in here (while trying to be thorough) I forgot that state is always going to be a str.